File: problem6.py
T = int(input())
for t in range(1, T+1):
    N, M = map(int, input().split())
    grid = [[-1 for _ in range(N)] for _ in range(N)]
    
    for i in range(N):
        grid[i] = list(map(int, input().split()))
    
    max_flies = 0
    
    for r in range(N):
        for c in range(N):
            sum_plus = grid[r][c]
            for step in range(1, M):
                nr_up = r - step
                if nr_up >=0:
                    sum_plus += grid[nr_up][c]
                nr_down = r + step
                if nr_down < N:
                    sum_plus += grid[nr_down][c]
                
                nc_left = c - step
                if nc_left >= 0:
                    sum_plus += grid[r][nc_left]
                nc_right = c + step
                if nc_right < N:
                    sum_plus += grid[r][nc_right]
            
            
            sum_x = grid[r][c]
            for step in range(1, M):
                nr_lu = r - step
                nc_lu = c - step
                if nr_lu >= 0 and nc_lu >=0:
                    sum_x += grid[nr_lu][nc_lu]
                
                nr_ld = r + step
                nc_ld = c - step
                if nr_ld < N and nc_ld >=0:
                    sum_x += grid[nr_ld][nc_ld]
                
                nr_ru = r - step
                nc_ru = c + step
                if nr_ru >= 0 and nc_ru < N:
                    sum_x += grid[nr_ru][nc_ru]
                
                nr_rd = r + step
                nc_rd = c + step
                if nr_rd < N and nc_rd < N:
                    sum_x += grid[nr_rd][nc_rd]
            
            max_flies = max(max_flies, sum_plus, sum_x)
    print(f"#{t} {max_flies}")

File: problem6.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;
    for (int t = 1; t<= T; t++){
        int N, M;
        cin >> N >> M;

        vector<vector<int>> grid(N, vector<int>(N));
        for(int i = 0; i< N; i++) {
            for(int j = 0; j < N; j++){
                cin >> grid[i][j];
            }
        }

        long long max_flies = 0;

        for (int r = 0; r< N; r++){
            for (int c = 0; c < N; c++){
                long long sum_plus = grid[r][c];
                for(int step = 1; step < M; step++){
                    
                    int nr_up = r - step;
                    if (nr_up >= 0) {
                        sum_plus += grid[nr_up][c];
                    }
                    
                    int nr_down = r + step;
                    if (nr_down < N) {
                        sum_plus += grid[nr_down][c];
                    }
                    
                    int nc_left = c - step;
                    if (nc_left >= 0) {
                        sum_plus += grid[r][nc_left];
                    }
                    
                    int nc_right = c + step;
                    if (nc_right < N) {
                        sum_plus += grid[r][nc_right];
                    }
                }

                long long sum_x = grid[r][c];
                for(int step = 1; step < M; step++) {
                    
                    int nr_lu = r - step;
                    int nc_lu = c - step;
                    if (nr_lu >= 0 && nc_lu >= 0) {
                        sum_x += grid[nr_lu][nc_lu];
                    }
                    
                    int nr_ru = r - step;
                    int nc_ru = c + step;
                    if (nr_ru >= 0 && nc_ru < N) {
                        sum_x += grid[nr_ru][nc_ru];
                    }
                    
                    int nr_ld = r + step;
                    int nc_ld = c - step;
                    if (nr_ld < N && nc_ld >= 0) {
                        sum_x += grid[nr_ld][nc_ld];
                    }
                    
                    int nr_rd = r + step;
                    int nc_rd = c + step;
                    if (nr_rd < N && nc_rd < N) {
                        sum_x += grid[nr_rd][nc_rd];
                    }
                }

                long long local_max = max(sum_plus, sum_x);
                if (local_max > max_flies){
                    max_flies = local_max;
                }

            }
        }

        cout << "#"<< t << " " << max_flies << "\n";

    }



    return 0;
}